packages/pigeon/lib/objc_generator.dart

```diff
@@ -67,9 +67,10 @@ String _className(String? prefix, String className) {
 
 /// Calculates callback block signature for for async methods.
 String _callbackForType(TypeDeclaration type, _ObjcPtr objcType) {
+  final String nullable = type.isNullable ? '_Nullable ' : '';
   return type.isVoid
       ? 'void(^)(NSError *_Nullable)'
-      : 'void(^)(${objcType.ptr.trim()}, NSError *_Nullable)';
+      : 'void(^)(${objcType.ptr.trim()}$nullable, NSError *_Nullable)';
```

Comment:
Doesn't this have to be nullable in all cases, just like the actual synchronous return value, for the case where there's an error?

Reply:
Yes, good call.